### DISCUSSION TITLE
take yaml on stdin

### DISCUSSION THREAD

**Discussion by aclater** [k9withabone/podlet#18] [Ideas]
It would be handy if podlet could take compose.yaml (or other) on stdin, so that one could use it in a pipeline of tools that output yaml or simply `cat compose.yaml | podlet`

**Comment by aclater** [comment]
Nevermind - I see this already works, you simply have to tell podlet what its taking in:

`cat compose.yaml | podlet compose`

**Comment by k9withabone** [comment]
> Nevermind - I see this already works

Not exactly, `podlet compose`, if you don't specify a file, will look for a compose file in the current directory. So in your given example podlet is just ignoring the stdin and finding the `compose.yaml` file in your current directory.
This behavior is documented in `podlet compose --help` in the `[COMPOSE_FILE]` section:
```
$ podlet compose --help

Generate podman quadlet files from a compose file

Creates a `.container` file for each service, a `.volume` file for each volume, and a `.network` file for each network.

The `--file` option must be a directory if used.

Some compose options are not supported, such as `build`.

When podlet encounters an unsupported option, an error will be returned. Modify the compose file to resolve the error.

Usage: podlet compose [OPTIONS] [COMPOSE_FILE]

Arguments:
  [COMPOSE_FILE]
          The compose file to convert
          
          If not provided, podlet will look for (in order) `compose.yaml`, `compose.yml`, `docker-compose.yaml`, and `docker-compose.yml`, in the current working directory.

Options:
      --pod <POD>
          Create a Kubernetes YAML file for a pod instead of separate containers
          
          A `.kube` file using the generated Kubernetes YAML file will also be created.

  -h, --help
          Print help (see a summary with '-h')
```
If your curious it is [implemented here](https://github.com/k9withabone/podlet/blob/ab526c1373d2f4fabd6c2321aaeef1364236f89d/src/cli.rs#L513).

I will add reading from stdin if the given path is `-` in the next version.

**Comment by aclater** [comment]
Ah! That explains it.

**Comment by k9withabone** [comment]
Implemented in v0.2.1
